compiler/front/options.nim

```diff
@@ -1162,16 +1164,12 @@ proc canonicalCase(path: var string) =
   else: toLowerAscii(path)
 
 proc fileInfoKnown*(conf: ConfigRef; filename: AbsoluteFile): bool =
-  var
-    canon: AbsoluteFile
-  try:
-    canon = canonicalizePath(conf, filename)
-  except OSError:
-    canon = filename
-  canon.string.canonicalCase
-  result = conf.m.filenameToIndexTbl.hasKey(canon.string)
+  result = conf.fileIdxTbl.hasKey(filename.string)
 
 proc fileInfoIdx*(conf: ConfigRef; filename: AbsoluteFile; isKnownFile: var bool): FileIndex =
+  let hasKey = conf.fileIdxTbl.hasKey(filename.string)
+  if hasKey:
+    return conf.fileIdxTbl[filename.string]
```

Comment:
```suggestion
  result = conf.fileIdxTbl.getOrDefault(filename.string, InvalidFileIdx)
  if result != InvalidFileIdx:
    return
```
This way it's slightly faster (one table lookup instead of two)